standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/SharedCache.java

```diff
@@ -1788,6 +2082,58 @@ public void addPartitionToCache(String catName, String dbName, String tblName, P
     }
   }
 
+  public void addPrimaryKeysToCache(String catName, String dbName, String tblName, List<SQLPrimaryKey> keys) {
+    try {
+      cacheLock.readLock().lock();
```

Comment:
We are writing in to cache. Check if we need read or write lock here? Looks like we take write lock on the table. Pls confirm if this is fine.

Reply:
This is fine, the below method (cachePrimaryKeys), takes the writeLock. the read lock here, is just to get the tblWrapper object.